### PR TITLE
arangodb: 3.4.8 -> 3.10.0

### PR DESCRIPTION
**Pull request by jsoo1**
###### Description of changes

`arangodb` was out of date (last commited 2018). This brings it to a more current version.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [X] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [X] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [X] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jsoo1** [comment]
Turns out `3.9.3 -> 3.10.0` wasn't too bad.

File: nixos/doc/manual/release-notes/rl-2211.section.md [hunk 1]
```diff
@@ -192,6 +192,8 @@ Available as [services.patroni](options.html#opt-services.patroni.enable).
 
 - The `aws` package has been removed due to being abandoned by the upstream. It is recommended to use `awscli` or `awscli2` instead.
 
+- `arangodb` versions 3.3, 3.4, and 3.5 have been removed because they are at EOL upstream. The default is now 3.10.0.
```
Comment (risicle) [inline nixos/doc/manual/release-notes/rl-2211.section.md:195]: We usually add a `throw` statement to `aliases.nix` for removed packages too.

File: pkgs/servers/nosql/arangodb/default.nix [hunk 2]
```diff
@@ -1,68 +1,60 @@
...
+, zlib
+, cmake
+, python3
+, perl
+, snappy
+, lzo
+, which
+}:
+
+gcc10Stdenv.mkDerivation rec {
```
Comment (risicle) [inline pkgs/servers/nosql/arangodb/default.nix:15]: There's an explicit warning in the cmake:

```
CMake Warning at CMakeLists.txt:616 (message):
  ArangoDB requires g++ 11.0 or newer, building with older compiler versions
  is unsupported
```

**Comment by yorickvP** [comment]
- #128338
- #128337
- #128336

File: pkgs/servers/nosql/arangodb/default.nix (hunk 2 above)
Comment (jsoo1) [reply pkgs/servers/nosql/arangodb/default.nix:15]: I see that but it does seem to compile fine. I'm not sure if they mean that gcc11 is required or that anything else is untested.

Comment (risicle) [reply pkgs/servers/nosql/arangodb/default.nix:15]: But why fix it to gcc10 in the first place? You haven't explained this.

**Comment by risicle** [comment]
Very unlikely, I think @yorickvP 's point is this will fix those issues.

**Comment by yorickvP** [comment]
Yep, this upgrade will fix those issues.

File: pkgs/servers/nosql/arangodb/default.nix (hunk 2 above)
Comment (jsoo1) [reply pkgs/servers/nosql/arangodb/default.nix:17]: You're right! Thank you! Added some commentary.

**Comment by jsoo1** [comment]
> Yep, this upgrade will fix those issues.

I see, thanks for the clarification!

File: pkgs/servers/nosql/arangodb/default.nix (hunk 2 above)
Comment (jsoo1) [reply pkgs/servers/nosql/arangodb/default.nix:17]: Do you think gcc10 is acceptable? I'm not sure what to make of the message (particularly the bit about being "unsupported")

Comment (yorickvP) [reply pkgs/servers/nosql/arangodb/default.nix:17]: gcc10 seems to work, but the point is, why override it from gcc11, which is the default in nixpkgs?

**Comment by vcunat** [comment]
That reasoning doesn't make sense to me.  You'd get gcc11 (which is supported by upstream).  Instead you force gcc10, with an argument speaking against such a change.

File: pkgs/servers/nosql/arangodb/default.nix (hunk 2 above)
Comment (jsoo1) [reply pkgs/servers/nosql/arangodb/default.nix:17]: Oh! It's just my error! I thought I had checked the stdenv.cc version and apparently I didn't read closely enough.  I'll change this today :)

**Comment by vcunat** [comment]
It (now) doesn't build for me on x86_64 NixOS.  Maybe there was some (other) reason to downgrade compiler version.  The errors look like changes around standard strictness, etc.  (complex C++ traces; probably no sense to post them here)

**Comment by jsoo1** [comment]
> It (now) doesn't build for me on x86_64 NixOS. Maybe there was some (other) reason to downgrade compiler version. The errors look like changes around standard strictness, etc. (complex C++ traces; probably no sense to post them here)

Same. I will investigate.

**Comment by jsoo1** [comment]
I found a new problem - maybe or maybe not related to compile failures - `/proc/cpuinfo` is read unconditionally for architecture feature detection. You turn this off by specifying `-DTARGET_ARCHITECTURE=xyz` up front. However, `-DTARGET_ARCHITECTURE=generic` seems like the best choice and that causes the vendored v8 not to compile.

**Comment by jsoo1** [comment]
I've asked upstream about the gcc 11 compile errors: https://github.com/arangodb/arangodb/issues/17454

**Comment by jsoo1** [comment]
I decided to keep only `x86_64-linux` as the supported system for `arangodb` since I can't find a reasonable default value for `-DTARGET_ARCHITECTURE` on that platform.  I documented the change and how to alter the new parameters.

**Comment by jsoo1** [comment]
This is probably about as much effort as I can give to this package at the moment. Hopefully what I have is sufficient but I would be happy to see some other eyes on this.

File: pkgs/servers/nosql/arangodb/default.nix [hunk 3]
```diff
@@ -1,68 +1,81 @@
...
-      '' else if (lib.versionOlder version "3.5") then ''
-        cp ${catch2}/include/catch2/catch.hpp 3rdParty/catch/catch.hpp
-      '' else ''
-        (cd 3rdParty/boost/1.69.0 && patch -p1 < ${../../../development/libraries/boost/pthread-stack-min-fix.patch})
-      ''}
-    '';
+  # prevent failing with "cmake-3.13.4/nix-support/setup-hook: line 10: ./3rdParty/rocksdb/RocksDBConfig.cmake.in: No such file or directory"
+  dontFixCmake = true;
+  NIX_CFLAGS_COMPILE = "-Wno-error";
+  preConfigure = "patchShebangs utils";
```
Comment (SuperSandro2000) [inline pkgs/servers/nosql/arangodb/default.nix:52]: This should really be a multiline string.

Comment (jsoo1) [reply pkgs/servers/nosql/arangodb/default.nix:52]: Thank you! https://github.com/NixOS/nixpkgs/pull/200210

**Pull request merged**